src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/ProjectEngineFactory.cs

```diff
@@ -12,10 +12,16 @@ internal abstract class ProjectEngineFactory : IProjectEngineFactory
 {
     protected abstract string AssemblyName { get; }
 
-    public RazorProjectEngine Create(
+    public abstract RazorProjectEngine Create(
         RazorConfiguration configuration,
         RazorProjectFileSystem fileSystem,
-        Action<RazorProjectEngineBuilder> configure)
+        Action<RazorProjectEngineBuilder> configure);
+
+    protected RazorProjectEngine Create(
+        RazorConfiguration configuration,
+        RazorProjectFileSystem fileSystem,
+        Action<RazorProjectEngineBuilder> configure,
+        bool registerCompilerFeatures)
```

Comment:
Rather than add a boolean parameter specifically for this, to control what code is a base class method does, perhaps adding a virtual method to this class, like `protected virtual void PreInitialize(RazorProjectEngineBuilder build) { }`, calling it before calling `initializer.Initialize()` below, and then overridding that in `ProjectEngineFactory_3_0.cs` would be more appropriate?

That way we don't get a proliferation of lots of parameters, and we can more easily compare to the legacy engines in case there are other differences in future

Reply:
Good idea, changed, thanks!